buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -141,34 +140,71 @@ class BuildPlugin implements Plugin<Project> {
+            File securityProperties = buildResources.copy("fips_java.security")
+            File security8Properties = buildResources.copy("fips_java8.security")
+            File securityPolicy = buildResources.copy("fips_java.policy")
+            File security8Policy = buildResources.copy("fips_java8.policy")
+            File bcfksKeystore = buildResources.copy("cacerts.bcfks")
+            GlobalInfoExtension globalInfo = project.rootProject.extensions.getByType(GlobalInfoExtension)
+            project.pluginManager.withPlugin("elasticsearch.testclusters") {
+                NamedDomainObjectContainer<ElasticsearchCluster> testClusters = project.extensions.findByName(TestClustersPlugin.EXTENSION_NAME) as NamedDomainObjectContainer<ElasticsearchCluster>
+                if (testClusters != null) {
+                    testClusters.all { ElasticsearchCluster cluster ->
+                        cluster.setTestDistribution(TestDistribution.DEFAULT)
+                        globalInfo.ready {
+                            for (File dep : project.getConfigurations().getByName("extraJars").getFiles()) {
+                                cluster.extraJarFile(dep)
                             }
+                            if (BuildParams.runtimeJavaVersion > JavaVersion.VERSION_1_8) {
+                                cluster.extraConfigFile("fips_java.security", securityProperties)
+                                cluster.extraConfigFile("fips_java.policy", securityPolicy)
+                            } else {
+                                cluster.extraConfigFile("fips_java.security", security8Properties)
+                                cluster.extraConfigFile("fips_java.policy", security8Policy)
+                            }
+                            cluster.extraConfigFile("cacerts.bcfks", bcfksKeystore)
+                            cluster.systemProperty('java.security.properties', '=${ES_PATH_CONF}/fips_java.security')
+                            cluster.systemProperty('java.security.policy', '=${ES_PATH_CONF}/fips_java.policy')
+                            cluster.systemProperty('javax.net.ssl.trustStore', '${ES_PATH_CONF}/cacerts.bcfks')
+                            cluster.systemProperty('javax.net.ssl.trustStorePassword', 'password')
+                            cluster.systemProperty('javax.net.ssl.keyStorePassword', 'password')
+                            cluster.systemProperty('javax.net.ssl.keyStoreType', 'BCFKS')
+                            // Can't use our DiagnosticTrustManager with SunJSSE in FIPS mode
+                            cluster.setting 'xpack.security.ssl.diagnose.trust', 'false'
                         }
                     }
                 }
+            }
+            project.tasks.withType(Test).configureEach { Test task ->
+                task.dependsOn(buildResources)
+                globalInfo.ready {
```

Comment:
Same as above, we can limit the surface area of `ready { }` here to only that bit conditional on java runtime version.